### PR TITLE
feat: refactor http request handlers

### PR DESCRIPTION
**Pull request by AllanFly120**
Replaces: https://github.com/aws/aws-sdk-js-v3/pull/1123
Resolves: #1107

Changes are made in the PR:
1. Previously, browsers' `fetch-http-handler` relies on the assumption that `Response.body` is a `ReadableStream`. This is not true for older desktop and mobile browsers. The previous design partially addresses this issue by explicitly notify `fetch-http-handler` to return body payload in`blob` instead of `ReadableStream` in ReactNative where `ReadableStream` is not available. But this solution doesn't solve proplems in old desktop browsers and IE.

    In this PR, the `fetch-http-handler` will dynamicly return body payload in `blob` or `ReadableStream` base on whether `Response.body` is implemented.  This solution return streaming payload data when it's available. It's also flexible as it will return streaming data if user supply a web stream polyfill.

2. Because we will use universal fetch handler in both browsers and ReactNative, we also merge the `stream-collector-browser` and `stream-collector-native` into a single `stream-collector-web`, which will collect the stream if payload's streaming, or convert the payload blob to binary data.

3. According to https://github.com/aws/aws-sdk-js-v3/pull/1123#issuecomment-627645312, I move the `stream-collector-web` into `fetch-http-handler` and move `stream-collector-node` into `node-http-handler` because  stream collector and request handler are intertwined with each other. It would prevent people incorrectly use node stream collector with browser request handler. Moreover, the 2 components will live in the same module but are different interfaces. So we can still introduce non-http handler without a stream collector in the future

/cc @russell-dot-js

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by AllanFly120** [comment]
✅integration test: 
```
AWS_PROFILE=js-sdk-team yarn test:integration
yarn run v1.22.1
$ cucumber-js --fail-fast
.................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

153 scenarios (153 passed)
534 steps (534 passed)
1m45.891s
Done in 109.77s.
```

**Comment by AllanFly120** [comment]
✅ReactNative iOS testing S3.listObjects

<details>
  <summary>expand to see the screen shot</summary>

Simulator Screen Shot - iPhone X - 2020-05-14 at 16 50 13

</details>

File: packages/fetch-http-handler/src/fetch-http-handler.spec.ts [hunk 1]
```diff
@@ -50,7 +50,8 @@ describe("httpHandler", () => {
     let response = await fetchHttpHandler.handle({} as any, {});
 
     expect(mockFetch.mock.calls.length).toBe(1);
-    expect(response.response.body).toBe("FOO");
+    console.log(response.response.body, typeof response.response.body);
```
Comment (trivikr) [inline packages/fetch-http-handler/src/fetch-http-handler.spec.ts:53]: nit: remove console.log
```suggestion
```

**Comment by Stereobit** [comment]
Thanks for this @AllanFly120 @russell-dot-js !

**Comment by russell-dot-js** [comment]
@AllanFly120 now that this is merged, how do we get a new release cut?
Also, for curiosity's sake, can you point out what changes you had to make for the codegen to properly generate the new clients?

**Pull request merged**